### ISSUE TITLE
endpoint for predictions

### ISSUE THREAD

**Issue by gyachdav** [PokemonGoers/PokeData#183]
Hello there pokedata, it has been a while 😆 

I didn't see any spec in your API that will allow the map team to retrieve predictions. Are predictions included in the sighting data? is there a parameter that distinguish predictions from sightings?

Same questions apply to the model that holds prediction data. 

Please respond ASAP.

**Comment by swathi-ssunder** [comment]
@gyachdav - Hello, the pokemon sightings from the API(that are fetched from the db) do not include the predictions.

**Comment by sacdallago** [comment]
Yes, because they should be generated on the fly :)

I guess once you can run the predictions, aka call the right function, and get an output, it will be pretty straightforward to implement the route.. Although: you could already implement the route, and imagine to have some magic `predictShit(lat, lng, timestamp)` function that returns the predictions and you later expose this to a GET request :)

**Comment by bensLine** [comment]
Btw. predictions for `predict(lat, lng, timestamp)` will be available as an array of the following objects:
`{"pokemonId":"16","confidence":"0.242","latitude":11.6088567,"longitude":48.1679286}`

**Comment by vivek-sethia** [comment]
Refer #187 . Changes will need to be made after the predictions are made available later.

**Comment by gyachdav** [comment]
referencing https://github.com/PokemonGoers/PredictPokemon-2/issues/67 again since I did not get any answer so far. 

According to @Aurel-Roci the integration of predictions into this project is complete. On the demo app I do not see any prediction. 

What is the status then? Is there a demo URL that will show that this integration is really done?

@PokemonGoers/pokedata @PokemonGoers/predictpokemon-2

**Comment by gyachdav** [comment]
@sacdallago please make sure this is handled in the next 24 hours thanks

**Comment by sacdallago** [comment]
Guys, can you give us an endpoint? :)

**Comment by vivek-sethia** [comment]
Referring to [this](https://github.com/PokemonGoers/PokeData/issues/183#issuecomment-253967793), the [route](http://pokedata.c4e3f8c7.svc.dockerapp.io:65014/doc/#api-PokemonSighting-GetPokemonPrediction) is already implemented.

As mentioned in the above comments, we only need to know which function needs to be invoked and were waiting on @PokemonGoers/predictpokemon-2 for this. 

Correct me if I am missing something here.

**Comment by Aurel-Roci** [comment]
The function is mentioned in the comments above: 
function:  predict(lat, lng, timestamp) 
response : {"pokemonId":"16","confidence":"0.242","latitude":11.6088567,"longitude":48.1679286}

**Comment by vivek-sethia** [comment]
@Aurel-Roci - And how are we supposed to include your project? Is it available as an npm package?

**Comment by Aurel-Roci** [comment]
@vivek-sethia Yes it is available as a package, I mentioned that in the issue with adding the java in the docker container. I thought this was already known by you guys since @jonas-he mentioned the package.

**Comment by vivek-sethia** [comment]
@Aurel-Roci - seems like some miscommunication. Will integrate the package and commit in a while.

**Comment by vivek-sethia** [comment]
@Aurel-Roci - I am trying to integrate and get the following error about the file `prediction_feature_config.json`.
Is there something that needs to be done additionally?

```
2016-10-28T18:05:13.742Z started prediction script, init DC
fs.js:634
  return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
                 ^

Error: ENOENT: no such file or directory, open 'prediction_feature_config.json'
    at Error (native)
    at Object.fs.openSync (fs.js:634:18)
    at Object.fs.readFileSync (fs.js:502:33)
    at fileToJson (PokeData/node_modules/predict-pokemon/dataSet_creator.js:364:23)
    at Object.DC.init (PokeData/node_modules/predict-pokemon/dataSet_creator.js:30:22)
    at PokeData/node_modules/predict-pokemon/prediction.js:31:8
    at Object.<anonymous> (PokeData/node_modules/predict-pokemon/prediction.js:241:3)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:456:32)
    at tryModuleLoad (module.js:415:12)
    at Function.Module._load (module.js:407:3)

```

**Comment by Aurel-Roci** [comment]
@vivek-sethia Yeah,  sorry forgot to tell you that you have to unzip `json/pokestop_groups.7z` file in the prediction npm package location. It will take some time to initialize the module. Let me know if it works

**Comment by vivek-sethia** [comment]
@Aurel-Roci - I already extracted it after referring to https://github.com/PokemonGoers/PredictPokemon-2/issues/67#issuecomment-254126234.

And still got this error, so wanted to know if there is anything else?

**Comment by Aurel-Roci** [comment]
@vivek-sethia I cant seem to reproduce that error. The only thing that comes to my mind is to add `/` before the file name `/prediction_feature_config.json`, like so.

**Comment by vivek-sethia** [comment]
@Aurel-Roci - Where exactly am I supposed to make this change ?

**Comment by Aurel-Roci** [comment]
@vivek-sethia  in node_modules/predict-pokemon/prediction.js. Line 31

**Comment by vivek-sethia** [comment]
@Aurel-Roci - Doesn't work. 
Well, do I have to copy the extracted `pokestop_groups.json` to some  other location?

**Comment by Aurel-Roci** [comment]
@vivek-sethia from what I see from the stacktrace the problem is with `prediction_feature_config.json`. For some reason it cannot find it, which I dont understand

**Comment by vivek-sethia** [comment]
@Aurel-Roci - Exactly, I already verified that the `prediction_feature_config.json` file exists at the expected location.
I have committed the [change](https://github.com/PokemonGoers/PokeData/commit/6ae99cce3ae05e333faeb617b83db7bd70b800ee). Maybe you could check once if there is something missing.

**Comment by Aurel-Roci** [comment]
@vivek-sethia  the `require('predict-pokemon')`, should be `require('predict-pokemon').predictor`.

**Comment by vivek-sethia** [comment]
@Aurel-Roci - Already tried that too. Same error as above.

**Comment by vivek-sethia** [comment]
@Aurel-Roci - Maybe you could try the code yourself and see it it works.

**Comment by semioniy** [comment]
@vivek-sethia I guess we found an error, stay tuned, we try to solve it.

**Comment by Aurel-Roci** [comment]
@vivek-sethia Thanks to @semioniy, this is what we found, first what has to be done is move the content of the  predict-pokemon package from Project/node_modules/predict-pokemon directly into the project directory. For some reason the paths do not seem to work correctly in the package. Also change the tzwhere like described here [https://www.npmjs.com/package/predict-pokemon](https://www.npmjs.com/package/predict-pokemon)

**Comment by vivek-sethia** [comment]
@Aurel-Roci Can you please make this change and test it yourself? Seems to be much easier that way. You can amend my commit if needed.

**Comment by Aurel-Roci** [comment]
@vivek-sethia @semioniy already tested it and it works

**Comment by vivek-sethia** [comment]
@Aurel-Roci @semioniy  Can I then merge my commit ?

**Comment by Aurel-Roci** [comment]
@vivek-sethia did you move the contents of the package directly to the project?

**Comment by vivek-sethia** [comment]
@Aurel-Roci - You mean in the commit? No, because we don't push the node_modules.

**Comment by Aurel-Roci** [comment]
@sacdallago can you help us with this problem. When the prediction package is published the paths seem not work properly. What is the cause of that?

**Comment by vivek-sethia** [comment]
@Aurel-Roci - Also, have you tested by including your npm package as a dependency in another project?

**Comment by semioniy** [comment]
@vivek-sethia that's what I did to test. It works after moving files to parent-parent directory, unzipping zip-files and changing tzwhere dependence. From a box it doesn't work.

**Comment by vivek-sethia** [comment]
@sacdallago - What do you suggest? Should we commit the entire `predict-pokemon` package in this repo too?

**Comment by vivek-sethia** [comment]
@sacdallago , @gyachdav - Any inputs on this?

**Comment by sacdallago** [comment]
Sorry guys I'm currently on a conference. I'll be able to give you valuable input tomorrow!

**Comment by jonas-he** [comment]
@sacdallago just reminding you that you wanted to give input

**Comment by sacdallago** [comment]
Thanks @jonas-he for reminding, sorry about that. Looks like a relative path problem, to me.

@semioniy @Aurel-Roci I've noticed a whole bunch of no-no's in your code:
1. https://github.com/PokemonGoers/PredictPokemon-2/blob/develop/prediction.js#L8 **NEVER** use constant URIs. That link is gonna be down in a couple days, what then? You should give a way of changing this via `init()` options.
2. https://github.com/PokemonGoers/PredictPokemon-2/blob/develop/prediction.js#L23 this is relative to the discussion here: **NEVER, EVER** load resources without specifying the relative file path! Always chain paths with the [`__dirname`](https://nodejs.org/api/globals.html#globals_dirname) through `path`. More at: http://stackoverflow.com/questions/8131344/what-is-the-difference-between-dirname-and-in-node-js
3. Point 2 will solve also the current issue, if you `path` chain https://github.com/PokemonGoers/PredictPokemon-2/blob/develop/prediction.js#L31

**Comment by vivek-sethia** [comment]
@Aurel-Roci , @semioniy , @sacdallago - Once the issues are fixed, we could merge #194  and then close this issue.

**Comment by Aurel-Roci** [comment]
@vivek-sethia The new package is released, you can install it and try it out. And let me now if it works

**Comment by vivek-sethia** [comment]
@sacdallago @semioniy @Aurel-Roci - Lots of errors after installing the new package and extracting the
`json/pokestop_groups.7z` file.
Some of the initial errors are below.
1. In `prediction.js`, File `prediction_feature_config.json` could not be found on line 41. This is because of a missing `/` before the file name.
2. After fixing the above error, similar error in file `dataSet_creator.js` on line 31.
3. After fixing the second error, file not found in `terrain_features.js` on line 6. This time, the file actually does not exist at the specified path but at a totally different location in `resources`.

Since the integration is already done in our project, I would suggest, that you fix all these issues and just change the package version in `package.json` of our project, test it and push it.

**Comment by sacdallago** [comment]
@Aurel-Roci @semioniy please fix this ASAP

**Comment by Aurel-Roci** [comment]
@vivek-sethia Now the package is working. Needs to be checked again when the API is up, but it should work then too

**Comment by sacdallago** [comment]
Mh @jonas-he can you check that this really is the case? If so, I will trigger a new docker deployment which will contain the update package

**Comment by swathi-ssunder** [comment]
@Aurel-Roci - It now refers to the latest version 0.1.6

**Comment by sacdallago** [comment]
Still not working:

``` bash
[pokedatadevelop-1]2016-11-07T10:16:17.644914864Z npm info it worked if it ends with ok
[pokedatadevelop-1]2016-11-07T10:16:17.645389310Z npm info using npm@2.14.2
[pokedatadevelop-1]2016-11-07T10:16:17.645692252Z npm info using node@v4.0.0
[pokedatadevelop-1]2016-11-07T10:16:19.096545617Z npm info prestart pokemongo-api@0.0.1
[pokedatadevelop-1]2016-11-07T10:16:19.162998917Z npm info start pokemongo-api@0.0.1
[pokedatadevelop-1]2016-11-07T10:16:19.170461598Z 
[pokedatadevelop-1]2016-11-07T10:16:19.170491476Z > pokemongo-api@0.0.1 start /usr/src/app
[pokedatadevelop-1]2016-11-07T10:16:19.170497735Z > NODE_ENV=production node app.js
[pokedatadevelop-1]2016-11-07T10:16:19.170502501Z 
[pokedatadevelop-1]2016-11-07T10:16:29.305042849Z 2016-11-07T10:16:29.271Z started prediction script, init DC
[pokedatadevelop-1]2016-11-07T10:16:51.054250201Z fs.js:549
[pokedatadevelop-1]2016-11-07T10:16:51.054310117Z   return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
[pokedatadevelop-1]2016-11-07T10:16:51.054321872Z                                                   ^
[pokedatadevelop-1]2016-11-07T10:16:51.054327505Z 
[pokedatadevelop-1]2016-11-07T10:16:51.054332621Z Error: ENOENT: no such file or directory, open '/usr/src/app/node_modules/predict-pokemon/feature_sources/../json/pokestop_groups.json'
[pokedatadevelop-1]2016-11-07T10:16:51.054338077Z     at Error (native)
[pokedatadevelop-1]2016-11-07T10:16:51.054343085Z     at Object.fs.openSync (fs.js:549:18)
[pokedatadevelop-1]2016-11-07T10:16:51.054347806Z     at Object.fs.readFileSync (fs.js:397:15)
[pokedatadevelop-1]2016-11-07T10:16:51.054352990Z     at fileToJson (/usr/src/app/node_modules/predict-pokemon/feature_sources/place_features.js:139:23)
[pokedatadevelop-1]2016-11-07T10:16:51.054358426Z     at /usr/src/app/node_modules/predict-pokemon/feature_sources/place_features.js:7:26
[pokedatadevelop-1]2016-11-07T10:16:51.054363620Z     at Object.<anonymous> (/usr/src/app/node_modules/predict-pokemon/feature_sources/place_features.js:142:3)
[pokedatadevelop-1]2016-11-07T10:16:51.054369031Z     at Module._compile (module.js:434:26)
[pokedatadevelop-1]2016-11-07T10:16:51.054373923Z     at Object.Module._extensions..js (module.js:452:10)
[pokedatadevelop-1]2016-11-07T10:16:51.054380581Z     at Module.load (module.js:355:32)
[pokedatadevelop-1]2016-11-07T10:16:51.054385643Z     at Function.Module._load (module.js:310:12)
[pokedatadevelop-1]2016-11-07T10:16:51.054390497Z     at Module.require (module.js:365:17)
[pokedatadevelop-1]2016-11-07T10:16:51.054395194Z     at require (module.js:384:17)
[pokedatadevelop-1]2016-11-07T10:16:51.054399834Z     at /usr/src/app/node_modules/predict-pokemon/dataSet_creator.js:52:30
[pokedatadevelop-1]2016-11-07T10:16:51.054465777Z     at Array.forEach (native)
[pokedatadevelop-1]2016-11-07T10:16:51.054475841Z     at Object.DC.init (/usr/src/app/node_modules/predict-pokemon/dataSet_creator.js:38:32)
[pokedatadevelop-1]2016-11-07T10:16:51.054481277Z     at /usr/src/app/node_modules/predict-pokemon/prediction.js:42:8
[pokedatadevelop-1]2016-11-07T10:16:51.339961005Z 
[pokedatadevelop-1]2016-11-07T10:16:51.358394394Z npm info pokemongo-api@0.0.1 Failed to exec start script
[pokedatadevelop-1]2016-11-07T10:16:51.359595861Z npm ERR! Linux 4.4.0-42-generic
[pokedatadevelop-1]2016-11-07T10:16:51.360807991Z npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
[pokedatadevelop-1]2016-11-07T10:16:51.361674069Z npm ERR! node v4.0.0
[pokedatadevelop-1]2016-11-07T10:16:51.369809860Z npm ERR! npm  v2.14.2
[pokedatadevelop-1]2016-11-07T10:16:51.371324803Z npm ERR! code ELIFECYCLE
[pokedatadevelop-1]2016-11-07T10:16:51.372038133Z npm ERR! pokemongo-api@0.0.1 start: `NODE_ENV=production node app.js`
[pokedatadevelop-1]2016-11-07T10:16:51.372839100Z npm ERR! Exit status 1
[pokedatadevelop-1]2016-11-07T10:16:51.374822582Z npm ERR! 
[pokedatadevelop-1]2016-11-07T10:16:51.375998085Z npm ERR! Failed at the pokemongo-api@0.0.1 start script 'NODE_ENV=production node app.js'.
[pokedatadevelop-1]2016-11-07T10:16:51.383628766Z npm ERR! This is most likely a problem with the pokemongo-api package,
[pokedatadevelop-1]2016-11-07T10:16:51.387430954Z npm ERR! not with npm itself.
[pokedatadevelop-1]2016-11-07T10:16:51.391594875Z npm ERR! Tell the author that this fails on your system:
[pokedatadevelop-1]2016-11-07T10:16:51.396719920Z npm ERR!     NODE_ENV=production node app.js
[pokedatadevelop-1]2016-11-07T10:16:51.406632817Z npm ERR! You can get their info via:
[pokedatadevelop-1]2016-11-07T10:16:51.410209885Z npm ERR!     npm owner ls pokemongo-api
[pokedatadevelop-1]2016-11-07T10:16:51.412017196Z npm ERR! There is likely additional logging output above.
[pokedatadevelop-1]2016-11-07T10:16:51.423986162Z 
[pokedatadevelop-1]2016-11-07T10:16:51.435966064Z npm ERR! Please include the following file with any support request:
[pokedatadevelop-1]2016-11-07T10:16:51.447907497Z npm ERR!     /usr/src/app/npm-debug.log
```

**Comment by swathi-ssunder** [comment]
@sacdallago - The above file not found error seems to be because the `json/pokestop_groups.7z` file is not unzipped.
@Aurel-Roci - Is it possible for you to directly include the `pokestop_groups.json` file?

**Comment by bensLine** [comment]
Great. What would life be without this constant predict pokemon struggle x)

Did you by chance not unzip the zips in the `data` and `json` directory?
`/usr/src/app/node_modules/predict-pokemon/feature_sources/../json/pokestop_groups.json` Is at least zipped if you just checked the package out.

**Comment by swathi-ssunder** [comment]
@bensLine - `node_modules` folder is not pushed to the repository. The packages are installed using the commands in the Dockerfile. (`npm install`).

**Comment by bensLine** [comment]
Oh, good :)
The file was to big for git, if there is no limitation for npm packages we should be able to do that.

**Comment by sacdallago** [comment]
@bensLine for the reference: you can use [unzip](https://www.npmjs.com/package/unzip) to unzip a zip file in your package before the packe is being installed, by defining this directive in the [preinstall script](https://docs.npmjs.com/misc/scripts) of the package.json. This is fancy AF. Or you can do the unzipping in prepublish, but that way you would have the unzipped files in your repo, although you could .gitignore them. There are PLENTY way to handle this :) but it should be done on your side!

**Comment by bensLine** [comment]
@sacdallago thanks for the info, I never used JS and npm before the seminar, so every pointer is very welcome :) someone of my team @PokemonGoers/predictpokemon-2 will look into it.

**Comment by sacdallago** [comment]
Since https://github.com/PokemonGoers/PredictPokemon-2/pull/74#event-856132523, @swathi-ssunder version bump, docker hub trigger: we see if it works?
@bensLine

**Comment by swathi-ssunder** [comment]
Updated to latest version #196. Hopefully everything works this time :pray:

**Comment by jonas-he** [comment]
@sacdallago its merged and docker build finished successfully. However DNS is not working right now. Maybe because there is still an error (->container shuts down) like we had in the past?

**Comment by sacdallago** [comment]
Merci! @jonas-he will have a look at the docker cloud log in the mornin'

**Comment by sacdallago** [comment]
Not over yet! 🗡 

``` bash
[pokedatadevelop-1]2016-11-13T01:16:24.075715817Z npm info it worked if it ends with ok
[pokedatadevelop-1]2016-11-13T01:16:24.077662800Z npm info using npm@2.14.2
[pokedatadevelop-1]2016-11-13T01:16:24.080686356Z npm info using node@v4.0.0
[pokedatadevelop-1]2016-11-13T01:16:25.659310042Z npm info prestart pokemongo-api@0.0.1
[pokedatadevelop-1]2016-11-13T01:16:25.708214646Z npm info start pokemongo-api@0.0.1
[pokedatadevelop-1]2016-11-13T01:16:25.716176367Z 
[pokedatadevelop-1]2016-11-13T01:16:25.716226527Z > pokemongo-api@0.0.1 start /usr/src/app
[pokedatadevelop-1]2016-11-13T01:16:25.716235727Z > NODE_ENV=production node app.js
[pokedatadevelop-1]2016-11-13T01:16:25.716241644Z 
[pokedatadevelop-1]2016-11-13T01:16:35.636328011Z 2016-11-13T01:16:35.633Z started prediction script, init DC
[pokedatadevelop-1]2016-11-13T01:16:51.489398555Z fs.js:549
[pokedatadevelop-1]2016-11-13T01:16:51.489440515Z   return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
[pokedatadevelop-1]2016-11-13T01:16:51.489452597Z                                                   ^
[pokedatadevelop-1]2016-11-13T01:16:51.489458631Z 
[pokedatadevelop-1]2016-11-13T01:16:51.489463581Z Error: ENOENT: no such file or directory, open '/usr/src/app/node_modules/predict-pokemon/feature_sources/../json/pokestop_groups.json'
[pokedatadevelop-1]2016-11-13T01:16:51.489468642Z     at Error (native)
[pokedatadevelop-1]2016-11-13T01:16:51.489473087Z     at Object.fs.openSync (fs.js:549:18)
[pokedatadevelop-1]2016-11-13T01:16:51.489477925Z     at Object.fs.readFileSync (fs.js:397:15)
[pokedatadevelop-1]2016-11-13T01:16:51.489482905Z     at fileToJson (/usr/src/app/node_modules/predict-pokemon/feature_sources/place_features.js:139:23)
[pokedatadevelop-1]2016-11-13T01:16:51.489488102Z     at /usr/src/app/node_modules/predict-pokemon/feature_sources/place_features.js:7:26
[pokedatadevelop-1]2016-11-13T01:16:51.489541341Z     at Object.<anonymous> (/usr/src/app/node_modules/predict-pokemon/feature_sources/place_features.js:142:3)
[pokedatadevelop-1]2016-11-13T01:16:51.489554023Z     at Module._compile (module.js:434:26)
[pokedatadevelop-1]2016-11-13T01:16:51.489559155Z     at Object.Module._extensions..js (module.js:452:10)
[pokedatadevelop-1]2016-11-13T01:16:51.489563271Z     at Module.load (module.js:355:32)
[pokedatadevelop-1]2016-11-13T01:16:51.489567232Z     at Function.Module._load (module.js:310:12)
[pokedatadevelop-1]2016-11-13T01:16:51.489571127Z     at Module.require (module.js:365:17)
[pokedatadevelop-1]2016-11-13T01:16:51.489575012Z     at require (module.js:384:17)
[pokedatadevelop-1]2016-11-13T01:16:51.489579314Z     at /usr/src/app/node_modules/predict-pokemon/dataSet_creator.js:52:30
[pokedatadevelop-1]2016-11-13T01:16:51.489583692Z     at Array.forEach (native)
[pokedatadevelop-1]2016-11-13T01:16:51.489588267Z     at Object.DC.init (/usr/src/app/node_modules/predict-pokemon/dataSet_creator.js:38:32)
[pokedatadevelop-1]2016-11-13T01:16:51.489593029Z     at /usr/src/app/node_modules/predict-pokemon/prediction.js:48:8
[pokedatadevelop-1]2016-11-13T01:16:51.641364127Z 
[pokedatadevelop-1]2016-11-13T01:16:51.650631189Z npm info pokemongo-api@0.0.1 Failed to exec start script
[pokedatadevelop-1]2016-11-13T01:16:51.651150535Z npm ERR! Linux 4.4.0-42-generic
[pokedatadevelop-1]2016-11-13T01:16:51.654334520Z npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
[pokedatadevelop-1]2016-11-13T01:16:51.655635753Z npm ERR! node v4.0.0
[pokedatadevelop-1]2016-11-13T01:16:51.657848977Z npm ERR! npm  v2.14.2
[pokedatadevelop-1]2016-11-13T01:16:51.659788031Z npm ERR! code ELIFECYCLE
[pokedatadevelop-1]2016-11-13T01:16:51.661028434Z npm ERR! pokemongo-api@0.0.1 start: `NODE_ENV=production node app.js`
[pokedatadevelop-1]2016-11-13T01:16:51.662301128Z npm ERR! Exit status 1
[pokedatadevelop-1]2016-11-13T01:16:51.663278508Z npm ERR! 
[pokedatadevelop-1]2016-11-13T01:16:51.664938311Z npm ERR! Failed at the pokemongo-api@0.0.1 start script 'NODE_ENV=production node app.js'.
[pokedatadevelop-1]2016-11-13T01:16:51.666558686Z npm ERR! This is most likely a problem with the pokemongo-api package,
[pokedatadevelop-1]2016-11-13T01:16:51.667410798Z npm ERR! not with npm itself.
[pokedatadevelop-1]2016-11-13T01:16:51.669910098Z npm ERR! Tell the author that this fails on your system:
[pokedatadevelop-1]2016-11-13T01:16:51.671229682Z npm ERR!     NODE_ENV=production node app.js
[pokedatadevelop-1]2016-11-13T01:16:51.674070397Z npm ERR! You can get their info via:
[pokedatadevelop-1]2016-11-13T01:16:51.675422797Z npm ERR!     npm owner ls pokemongo-api
[pokedatadevelop-1]2016-11-13T01:16:51.677335732Z npm ERR! There is likely additional logging output above.
[pokedatadevelop-1]2016-11-13T01:16:51.690191312Z 
[pokedatadevelop-1]2016-11-13T01:16:51.694651110Z npm ERR! Please include the following file with any support request:
[pokedatadevelop-1]2016-11-13T01:16:51.704430089Z npm ERR!     /usr/src/app/npm-debug.log
```

**Comment by sacdallago** [comment]
@bensLine @Aurel-Roci

**Comment by Aurel-Roci** [comment]
@sacdallago did you call unzipFiles() first? Like here https://www.npmjs.com/package/predict-pokemon. If it still works after that, I can do something with callbacks, but I would like your feedback on it.

**Comment by swathi-ssunder** [comment]
@sacdallago , @Aurel-Roci - Even after `unzipFiles()`, the status is same as in https://github.com/PokemonGoers/PokeData/issues/183#issuecomment-260123638

**Comment by bensLine** [comment]
@Aurel-Roci I added a post install script to unzip the files. can you please release the package again to test it. https://github.com/PokemonGoers/PredictPokemon-2/pull/75

**Comment by Aurel-Roci** [comment]
@swathi-ssunder  the new version is up

**Comment by jonas-he** [comment]
see #198

**Comment by swathi-ssunder** [comment]
@Aurel-Roci , @bensLine 

```
Error: Cannot find module 'fstream'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:286:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/usr/src/app/node_modules/predict-pokemon/scripts/postinstall.js:3:15)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:475:10)
npm info predict-pokemon@0.2.3 Failed to exec postinstall script
npm ERR! Linux 3.19.0-66-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v4.0.0
npm ERR! npm  v2.14.2
npm ERR! code ELIFECYCLE

npm ERR! predict-pokemon@0.2.3 postinstall: `node scripts/postinstall.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the predict-pokemon@0.2.3 postinstall script 'node scripts/postinstall.js'.
npm ERR! This is most likely a problem with the predict-pokemon package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node scripts/postinstall.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls predict-pokemon
npm ERR! There is likely additional logging output above.
npm info preuninstall predict-pokemon@0.2.3
npm info uninstall predict-pokemon@0.2.3
npm info postuninstall predict-pokemon@0.2.3
npm ERR! Please include the following file with any support request:
npm ERR!     /usr/src/app/npm-debug.log
The command '/bin/sh -c npm install' returned a non-zero code: 1
The command "docker build -t pokemongoers/pokedata ." failed and exited with 1 during .
```

**Comment by Aurel-Roci** [comment]
@swathi-ssunder package dependency was missing, added now. Sorry for all this

**Comment by swathi-ssunder** [comment]
@sacdallago - The docker build was successful this time, the DNS still does not work though!

**Comment by jonas-he** [comment]
so i looked into this and it turns out that after executing `npm start` this is the output:

```
npm info it worked if it ends with ok
npm info using npm@2.14.2
npm info using node@v4.0.0
npm info prestart pokemongo-api@0.0.1
npm info start pokemongo-api@0.0.1

> pokemongo-api@0.0.1 start /usr/src/app
> NODE_ENV=production node app.js

2016-11-16T16:39:27.282Z started prediction script, init DC
4837ms to construct shortcut table
Average timezones per 1° lat x 1° lng: 4.414128352490422
2016-11-16T16:39:35.486Z created training arff file
2016-11-16T16:39:35.487Z requesting http://predictemall.online:65014/api/pokemon/sighting/ts/2016-09-14T08:00:00.000Z/range/1d?token=I0TPIIpCLH8lR8iDrCMV
npm info poststart pokemongo-api@0.0.1
npm info ok 
```

after this the process exits. Dont know exactly whats causing this. might be that it exits because the request to the predictemall site fails (i get a bad gateway when trying it in browser).

**Comment by jonas-he** [comment]
sorry it turns out that my docker container exits because it couldnt connect to the database. It was still configured with mlab database, but i guess it was taken down, so authentication error -> process exits. To know what the real problem is well have to wait on @sacdallago

**Comment by sacdallago** [comment]
It seems to be up and running to me :) Can you confirm that the predictions are visible?

**Comment by jonas-he** [comment]
@sacdallago are you talking about the dockerhub thing or predictemall.online ? The first is down for me the latter is up. However I see no possibility to request predictions

**Comment by sacdallago** [comment]
@jonas-he 😮 how didn't I reply to that. Sorry :(

Yes, yes. Everything is up where it should be, and down where it should be down :) Why don't you see a possibility to request predictions?

**Comment by jonas-he** [comment]
@sacdallago i can turn prediction toggle on/off and change the time range but nothing happens on the map as well as in terms of network requests (i see requests for sightings but not for predictions).

Another thing is that there is still some dummy running, which is generating tons of pokemobs in munich.

**Comment by samitsv** [comment]
updated the api for prediction, https://github.com/PokemonGoers/PokeData/pull/199, but one of the build fails https://travis-ci.org/PokemonGoers/PokeData/jobs/184915843
